src/tslint.ts

```diff
@@ -43,9 +44,20 @@ module Lint {
             this.fileName = fileName;
             this.source = source;
             this.options = options;
+
+            if (this.options.allowNonTsExtensions == null) {
+                this.options.allowNonTsExtensions = false;
+            }
         }
 
         public lint(): LintResult {
+            const tsExtensions = [".ts"];
+            const fileExtension = path.extname(this.fileName);
+            const isTsFile = (tsExtensions.indexOf(fileExtension) >= 0);
```

Comment:
no need for the surrounding parens here. also let's inline the `fileExtensions` variable